content/_code-samples/non-fungible-token/py/mint-nft.py

```diff
@@ -0,0 +1,66 @@
+from xrpl.transaction import safe_sign_and_autofill_transaction, send_reliable_submission
+from xrpl.models.transactions.nftoken_mint import NFTokenMint, NFTokenMintFlag
+from xrpl.wallet import generate_faucet_wallet
+from xrpl.models.requests import AccountNFTs
+from xrpl.clients import JsonRpcClient
+from xrpl.wallet import Wallet
+
+# Mint an NFT on the XRPL via a NFTokenMint transaction
+# https://xrpl.org/nftokenmint.html#nftokenmint
+
+# If you want to mint a NFT on an already existing account, enter in the seed. If not, an account will be provided
+# Make sure the seed variable is empty "", if you want to use a brand new testing account
+
+seed = ""
+
+# Connect to a testnet node
+print("Connecting to Testnet...")
+JSON_RPC_URL = "https://s.altnet.rippletest.net:51234/"
+client = JsonRpcClient(JSON_RPC_URL)
+
+# Get issuer account credentials from the testnet faucet
+if seed == "":
+    print("Requesting address from the Testnet faucet...")
+    issuer_wallet = generate_faucet_wallet(client=client)
+    issuerAddr = issuer_wallet.classic_address
+else:
+    issuer_wallet = Wallet(seed=seed, sequence=0)
+    issuerAddr = issuer_wallet.classic_address
+
+print(f"\nIssuer Account: {issuerAddr}")
+print(f"          Seed: {issuer_wallet.seed}")
+
+# Construct NFTokenMint transaction to mint 1 NFT
+mint_tx = NFTokenMint(
+    account=issuerAddr,
+    nftoken_taxon=1,
+    flags=NFTokenMintFlag.TF_TRANSFERABLE
+)
+
+# Sign mint_tx using the issuer account
+mint_tx_signed = safe_sign_and_autofill_transaction(transaction=mint_tx, wallet=issuer_wallet, client=client)
+mint_tx_signed = send_reliable_submission(transaction=mint_tx_signed, client=client)
+mint_tx_result = mint_tx_signed.result
+
+print(f"\n  Mint tx result: {mint_tx_result['meta']['TransactionResult']}")
+print(f"      Tx content: {mint_tx_result}")
+
+for node in mint_tx_result['meta']['AffectedNodes']:
+    if "CreatedNode" in list(node.keys())[0]:
+        print(f"\n - NFT metadata:"
+              f"\n        NFT ID: {node['CreatedNode']['NewFields']['NFTokens'][0]['NFToken']['NFTokenID']}"
+              f"\n  Raw metadata: {node}")
```

Comment:
Unfortunately this doesn't quite work if you repeatedly mint on the same account. (I believe this is because it only makes a new NFTokenPage on the 1st NFT, and every 15-30 NFTs after that) The information to find the NFTokenID is still there, but you have to navigate modified nodes & such, which can get quite complicated. I created an issue internally to add a helper to handle that in the near future called something like `getNFTokenIDFromResponse` potentially.

Reply:
Right, we'll stick with my code until this feature exist. Thanks!